File: Social-Media-Backend/ChattingApp/src/main/java/com/ChattingApp/DAO/UserDAO.java
package com.ChattingApp.DAO;

import com.ChattingApp.Entity.User;

public interface UserDAO {
	
	void deleteByEmail(String email);
	
	User updateUser(User user);
	
	User findByEmail(String email);
	
	void saveUser(User user);
	
	void updateIsVerifiedEmailByEmail(String email, boolean isVerifiedEmail);
	
	

	
}

File: app/src/main/java/com/chattingapp/chattingapp/dao/UserDAO.kt
package com.chattingapp.chattingapp.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import com.chattingapp.chattingapp.constants.Constants
import com.chattingapp.chattingapp.entity.User

@Dao
interface UserDAO {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUser(user: User)

    @Update
    fun updateUser(user: User)

    @Query("DELETE FROM ${Constants.TABLE_NAME}")
    fun deleteAllUsers()

    @Query("SELECT * FROM ${Constants.TABLE_NAME} WHERE email =:email")
    fun getUserByEmail(email:String):User

    @Query("SELECT * FROM ${Constants.TABLE_NAME} ORDER BY email ASC")
    fun getAllUsers(): LiveData<List<User>>


}